core/tx_pool.go

```diff
@@ -630,7 +630,7 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (bool, error) {
 		return false, err
 	}
 	// If the transaction pool is full, discard underpriced transactions
-	if uint64(pool.all.Count()) >= pool.config.GlobalSlots+pool.config.GlobalQueue {
+	if !local && uint64(pool.all.Count()) >= pool.config.GlobalSlots+pool.config.GlobalQueue {
```

Comment:
I'm unsure if this is the correct solution. This code path not only handles adding a transaction, but also removing ones that got out-priced.  Your modification breaks the case where the pool is full but you add a local transaction. Originally that discarded a non-local one, whereas your new case keeps the non-local too.

Reply:
I have checked the codes that remove ones that got out-priced. I found that it never remove local txs actually. 
So I think it's right, but it's truly ugly because of the implicit association.

Reply:
@karalabe you're right. This modify breaks the originally logic. I'll dismiss this PR.